controllers/common_test.go

```diff
@@ -36,15 +36,25 @@ var _ = Describe("GetGeneratedJobName", func() {
 		uidWithoutHyphens string
 
 		generatedJobName string
+		generatedPostfix string
+
+		// If this is true, a postfix is added to the job name
+		shouldJobNameHavePostfix bool
 	)
 
 	BeforeEach(func() {
 		uid = types.UID(uuid.New().String())
+		shouldJobNameHavePostfix = false
 	})
 
 	JustBeforeEach(func() {
 		uidWithoutHyphens = strings.Replace(string(uid), "-", "", -1)
-		generatedJobName = GetGeneratedJobName(uid, objectMetaName, maxNameLen)
+
+		if shouldJobNameHavePostfix {
+			generatedJobName = GetGeneratedResourceName(generatedPostfix+"-"+uidWithoutHyphens, objectMetaName, maxNameLen)
+		} else {
+			generatedJobName = GetGeneratedJobName(uid, objectMetaName, maxNameLen)
```

Comment:
Why are we testing another function in `GetGeneratedResourceName` tests? This couples `GetGeneratedResourceName` with `GetGeneratedJobName`, making code harder to refactor and tests more difficult to comprehend.

I think since `GetGeneratedJobName` merely calls `GetGeneratedResourceName`, we only need a test on the calling logic for `GetGeneratedJobName`. Can we break out a new `Describe` case for `GetGeneratedJobName` which only tests 1) that the UID does not have hyphens and 2) that the output is equivalent to `GetGeneratedResourceName` output for some basic case?